core/dbt/events/types.py

```diff
@@ -486,6 +486,16 @@ def message(self) -> str:
         return line_wrap_message(warning_tag(description))
 
 
+class MicrobatchMacroOutsideOfBatchesDeprecation(WarnLevel):
+    def code(self) -> str:
+        return "D020"
+
+    def message(self) -> str:
+        description = "The use of a custom microbatch macro outside of batched execution is deprecated. To use it with batched execution, set `flags.require_batched_execution_for_custom_microbatch_strategy` to `True` in `dbt_project.yml`. In the future this will be the default behavior."
```

Comment:
super nit, just because this is user-facing: "In the future this will be the default behavior." -> "In the future batched execution will be the default behaviour."

Reply:
We could also mention the name of the custom macro that exists (e.g. `get_incremental_microbatch_sql` as part of the message in case the user that's seeing this doesn't know about the custom strategy macro naming convention offhand.